### ISSUE TITLE
flake registry could record dir path as well

### ISSUE THREAD

**Issue by kquick** [NixOS/nix#4050]
Currently a `nix registry add flake:foo github:team/repo?dir=subdir/in/repo` ignores the dir specification and just registers the repo itself.  This means that when used, the current syntax should be `inputs.foo.url = flake:foo?dir=subdir/in/repo` to properly reference the desired location.

The suggested enhancement is to allow the registry to record the dir specification as well.  This would allow the registry to be used to absorb any future changes in the corresponding repos.  For example, if `subdir/in/repo/flake.nix` became `nix/flake.nix`, this could be updated by the single change updating the registry for the `flake:foo` rather than editing every flake that imports `flake.foo` and changing the directory path in those individual flakes.

**Comment by edolstra** [comment]
The registry does support subdirectories, see for instance [`blender-bin`](https://github.com/NixOS/flake-registry/blob/090d2e959af4b35d35d7b0717c5669e77ba9678b/flake-registry.json#L4-L13). Maybe `nix registry add` is dropping the `dir`?

**Comment by cole-h** [comment]
What happens if you quote the URI (e.g. `nix registry add flake:foo 'github:team/repo?dir=subdir/in/repo'`)? Maybe your shell is doing something funky with the `?` that causes it to ignore everything after.

**Comment by kquick** [comment]
Sorry for the confusion, there were multiple effects occurring here:
1. The directory path isn't currently visible in the `registry list` output.
2. I had some automation around this that was choosing  between a github: or path: target for the registry entry and it was specifying `$ nix registry add flake:foo /path/to/root?dir=sub/path`, which is accepted, but the attribute portion is silently dropped.
3. The result was invalid, so a `$ nix flake info flake:foo` displayed only an error.

I've added a pull-request to display the attribute info on the `registry list` command to help with the first item and hopefully avoid others getting confused by this.

P.S. Thanks for the suggestion @cole-h... I had tried that as well, and it didn't help.  It turns out that `nix` is the one ignoring the `?` and everything following it.  :-)
